pallets/dapp-staking-v3/src/lib.rs

```diff
@@ -478,6 +479,7 @@ pub mod pallet {
         pub tier_thresholds: Vec<TierThreshold>,
         pub slots_per_tier: Vec<u16>,
         pub _config: PhantomData<T>,
+        pub disable_safeguard: bool,
```

Comment:
If the default value of `Safeguard` changes, does this approach still work?
E.g. `Safeguard` is set to `false` by default, and user sets `disable_safeguard` to `false`. What's the result?

Perhaps it's better to set the storage value directly.

Nitpick comment - `PhantomData` should be at the bottom of the fields declaration.